pkg/distributor/query.go

```diff
@@ -191,28 +192,40 @@ func (d *Distributor) queryIngesterStream(ctx context.Context, replicationSet ri
 	queryLimiter := limiter.QueryLimiterFromContextWithFallback(ctx)
 	reqStats := stats.FromContext(ctx)
 
-	queryIngester := func(ctx context.Context, ing *ring.InstanceDesc, cleanup context.CancelFunc) (ingesterQueryResult, error) {
-		client, err := d.ingesterPool.GetClientFor(ing.Addr)
-		if err != nil {
-			return ingesterQueryResult{}, err
-		}
-
-		stream, err := client.(ingester_client.IngesterClient).QueryStream(ctx, req)
-		if err != nil {
-			return ingesterQueryResult{}, err
+	queryIngester := func(ctx context.Context, ing *ring.InstanceDesc, cleanupContext context.CancelFunc) (ingesterQueryResult, error) {
```

Comment:
[nit] Maybe `cancelContext` is more clean than `cleanupContext`?

Reply:
Makes sense, changed in 920144c